var/spack/repos/builtin/packages/mvapich2-gdr/package.py

```diff
@@ -0,0 +1,191 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import sys
+from spack import *
+
+class Mvapich2Gdr(AutotoolsPackage):
+    # Mvapich2-Gdr is not installable from source and is only available through the binary mirror below
+    # http://mvapich.cse.ohio-state.edu:8080/download/mvapich/spack-mirror/mvapich2-gdr/
+
+    homepage = "http://mvapich.cse.ohio-state.edu"
+    url      = "http://mvapich.cse.ohio-state.edu:8080/download/mvapich/spack-mirror/mvapich2-gdr/mvapich2-gdr-2.3.4.tar.gz"
+    
+    version('2.3.4', sha256='ed78101e6bb807e979213006ee5f20ff466369b01f96b6d1cf0c471baf7e35aa')
+    version('2.3.3', sha256='9b7b5dd235dbf85099fba3b6f1ccb49bb755923efed66ddc335921f44cb1b8a8')
+    
+    provides('mpi')
+    provides('mpi@:3.1')
+
+    variant(
+        'cuda_version',
+        description='Currently supported for CUDA versions 9.2, 10.1 and 10.2',
+        default='10.1',
+        values=('10.2', '10.1', '9.2'),
+        multi=False
+    )
```

Comment:
No need for a variant for this, Spack already tracks the versions of every dependency in the DAG:
```console
$ spack install mvapich2-gdr ^cuda@10.1.243
```

Reply:
I was forced to add a "wrapper" around the cuda variant because I didn't find a way to list the dependencies of the binaries that were available in `$ spack buildcache list`

`$ spack find` has a `-d` option so the user can know the version of dependencies but i found no equivalent for `spack buildcache list`

So to be able to specify what the allowed values of cuda are(the versions that we have generated binaries for), I added the redundant variant.

This way, I was able to add a description in the variant field, so users will know what cuda versions are acceptable in `$ spack info`

Reply:
You can specify supported versions with:
```python
depends_on('cuda@9.2:10.2')
```
As for the buildcache issue, feel free to open a separate issue for that. @gartung may be able to quickly add the feature you need.

Reply:
Use `spack buildcache list -lv` to list the buildcaches available including the variants used and the hash.
Use `spack buildcache install /hash` to install the package and all of it's link and run dependencies. 
The hash is shorthand for the full spec.

Reply:
@gartung We have created our binaries with `spack buildcache create --only package`, so spack buildcache install /hash fails for me. Is there any way to get it to install dependencies from source but the binary from the mirror using only `$ spack buildcache install /hash`

Reply:
That can't be fixed unless you also created buildcaches for the dependencies.

Reply:
Currently making this change and generating the appropriate binaries. Will push the new commits within 24 hours.

Reply:
> You can specify supported versions with:
> 
> ```python
> depends_on('cuda@9.2:10.2')
> ```
> 
> As for the buildcache issue, feel free to open a separate issue for that. @gartung may be able to quickly add the feature you need.

I have raised this feature request: [link](https://github.com/spack/spack/issues/17998#issue-677240730)

Reply:
@adamjstewart I've addressed your requested changes in this commit:- 97fcf839338c83b6fb9bab63e8702adbf81bb119